zkvm/src/vm.rs

```diff
@@ -281,6 +282,31 @@ where
         Ok(())
     }
 
+    fn r#const(&mut self) -> Result<(), VMError> {
+        let data = self.pop_item()?.to_data()?.to_bytes();
+        let mut slice = Subslice::new(&data);
+        let scalar = slice.read_scalar()?;
```

Comment:
```suggestion
        let scalar = slice.read_scalar()?;
        if slice.len() != 0 { return Err(VMError::FormatError) }
```

We should not allow garbage in the trailing bytes.

Reply:
This will be simplified and made safer later: https://github.com/interstellar/slingshot/issues/80